gui/builtinViews/fittingView.py

```diff
@@ -408,6 +408,9 @@ def swapItems(self, x, y, srcIdx):
             if mod1.slot != mod2.slot:
                 return
 
+            if not getattr(mod2, "modPosition"):
+                self.mods.modPosition = mod2.modPosition = dstRow
+
```

Comment:
>When will modPosition be empty? If we're in the swapItems methods, then that means we have two modules to swap and they will both have this property simply due to the fact that they are active modules. It's okay to assume we return something here, unless you noted it actually breaking during normal use. In which case, if they don't have it, I'd like for it to break so we know about it rather than trying to deduce which position the module is supposed to be in.

It's more of a "this shouldn't happen but it could" scenario.

Honestly, I can't remember if I actually ran into this breaking.

>It's okay to assume we return something here

Defensive programming all the way! :)

>Furthermore, we shouldn't be setting modPosition. It doesn't have a setter and shouldn't have one - this is supposed to be a read-only property that returns the modules index in the fit's module list. I think this is being done all wrong.
>Additionally, I'm curious on two things:
>Why are you setting self.mods.modPosition? self.mods is a list of modules...

Dammit, I can't hide stupid code on git. :(

>Why getattr(mod2, "modPosition") instead of simply mod2.modPosition?

Now this one is legitimate and half right.  Because in Python `None` = `False` = `0`.

Anyway, it's that last one that will get you, since often lists and positions start with 0.  Rather than assuming that `modPosition` can never have a position of 0, we use `getattr` which will return nothing (since no default is set), and not run into problems if it's legitimately set to 0.

Of course in this case if `modPosition` were actually 0....ummm....so yeah, I ended doing the same thing by trying to avoid it.

We can avoid it entirely by using `is not None` and `is not False` because `is` is explicit.  But I think what I changed it to avoids this entirely.